src/tests/baseservices/exceptions/stackoverflow/stackoverflowtester.cs

```diff
@@ -3,26 +3,82 @@
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
 using System.IO;
+using System.Linq;
 using System.Text;
 using Xunit;
 
 namespace TestStackOverflow
 {
     public class Program
     {
-        static void TestStackOverflow(string testName, string testArgs, out List<string> stderrLines)
+        const string EnableMiniDumpEnvVar = "DOTNET_DbgEnableMiniDump";
+
+        public static int Main(string[] args)
+        {
+            if (args.Length > 0)
+            {
+                string testName = args[0];
+                string[] testArgs = Array.Empty<string>();
+                switch (testName)
+                {
+                    case "stackoverflow":
+                        Assert.True(args.Length >= 2);
+                        testArgs = args[1..].ToArray();
+                        StackOverflow.Run(testArgs);
+                        break;
+                    case "stackoverflow3":
+                        StackOverflow3.Run();
+                        break;
+                    default:
+                        throw new InvalidOperationException($"Invalid arguments to 'stackoverflowtester' process. Test '{testName}' does not exist");
+                }
+                throw new InvalidOperationException($"Invalid arguments to 'stackoverflowtester' process. Test '{testName}' with arguments '{testArgs}' should have thrown an exception.");
+            }
+
+            int exitCode = 101;
+            try
+            {
+                TestStackOverflowSmallFrameMainThread();
+                exitCode++;
+                TestStackOverflowLargeFrameMainThread();
+                exitCode++;
+                TestStackOverflowSmallFrameSecondaryThread();
+                exitCode++;
+                TestStackOverflowLargeFrameSecondaryThread();
+                exitCode++;
+                TestStackOverflow3();
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e.ToString());
+                return exitCode;
+            }
+
+            return 100;
+        }
+
+        [UnconditionalSuppressMessage("SingleFile", "IL3000", Justification = "We want an empty string for location if the test is running as single file")]
+        static bool TestStackOverflow(string testName, string testArgs, out string[] stderrLines, out bool checkStackFrame)
         {
-            Console.WriteLine($"Running {testName} test({testArgs})");
             List<string> lines = new List<string>();
 
-            Process testProcess = new Process();
+            string thisAssemblyPath = typeof(Program).Assembly.Location;
+            // TestLibrary.Utilities.IsNativeAot brings in ~4mb on Linux, so we'll approximate with this for now. https://github.com/dotnet/runtime/issues/94313
+            bool IsNativeAot = string.IsNullOrEmpty(thisAssemblyPath) && !RuntimeFeature.IsDynamicCodeSupported;
```

Comment:
I think you _actually_ want to test for single file, not for Native AOT. If we ever add single file (non-native AOT) runs of the test tree, this code would need tweaking.

Might as well just test for single file.

https://github.com/dotnet/runtime/blob/9d3f324640e8083ddfaebeed91119cd69d43cd33/src/tests/Common/CoreCLRTestLibrary/Utilities.cs#L100